File: netlogo-gui/src/main/window/ClientAppInterface.java
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.window;

import org.nlogo.api.CompilerServices;
import org.nlogo.nvm.PresentationCompilerInterface;

public interface ClientAppInterface {
  void startup(final String userid, final String hostip,
               final int port, final boolean isLocal, final boolean isRobo, final long waitTime,
               final PresentationCompilerInterface workspace);
}

File: netlogo-gui/src/main/window/ClientAppInterface.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.window

import java.awt.Window

import org.nlogo.api.{ CompilerServices, ExtensionManager }
import org.nlogo.theme.ThemeSync

trait ClientAppInterface extends Window with ThemeSync {
  def startup(userid: String, hostip: String, port: Int, isLocal: Boolean, isRobo: Boolean, waitTime: Long,
              workspace: CompilerServices, extensionManager: ExtensionManager): Unit
}
